packages/top-app-bar/index.js

```diff
@@ -48,11 +48,17 @@ export default class TopAppBar extends React.Component {
     this.foundation_.init();
   }
 
-  addClassesToElement(classes, element) {
-    const propsWithClasses = {
+  addClassesAndEnableRippleOnElement(classes, element) {
+    const updatedProps = {
       className: classnames(classes, element.props.className),
     };
-    return React.cloneElement(element, propsWithClasses);
+
+    // If `element` is a Native React Element, we don't want to throw
+    // a warning.
```

Comment:
I think we should have a failure if you give us a Native React Element. The child needs to implement the `hasRipple` property, so it has to be a proper React.Component

Reply:
The Native React Components are still React Components. I don't think we want to be that restrictive here. We spent time in the TopAppBar PR to not limit this.